client/allocrunner/networking_bridge_linux_test.go

```diff
@@ -0,0 +1,45 @@
+package allocrunner
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_buildNomadBridgeNetConfig(t *testing.T) {
+	testCases := []struct {
+		name string
+		b    *bridgeNetworkConfigurator
+	}{
+		{
+			name: "empty",
+			b:    &bridgeNetworkConfigurator{},
+		},
+
+		{
+			name: "hairpin",
+			b: &bridgeNetworkConfigurator{
+				bridgeName:  defaultNomadBridgeName,
+				allocSubnet: defaultNomadAllocSubnet,
+				hairpinMode: true,
+			},
+		},
+		{
+			name: "bad_name",
+			b: &bridgeNetworkConfigurator{
+				bridgeName: `bad","foo":"bar`,
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc := tc
+			var out []byte
+			require.NotPanics(t, func() { out = buildNomadBridgeNetConfig(*tc.b) })
+			if tc.name == "bad_name" {
+				fmt.Println(string(out))
```

Comment:
I'm curious why print the output for "bad_name" ?  and not any others?

and I find myself wanting to assert more than just "doesn't panic" but on the fence about what specifically to suggest short of full-text string matching...

Reply:
The bad_name test was to see if JSON stuffing was a concern. While it appears to be, it will cause the [`generateAdminChainRule` function](https://github.com/hashicorp/nomad/blob/778ab0fcd6d8f76ce8a03ed7cb37bea4bc0af47c/client/allocrunner/networking_bridge_linux.go#L121-L125) to break.  This PR doesn't actually introduce the JSON stuffing issue and ultimately, bridge names would benefit from some additional validation since they aren't built until they are needed the first time.

Reply:
I removed the test since it doesn't actually test whether or not the template _rendering_ breaks.  Even in the original code, we dropped in `%s` inside of quotes rather than leveraging `%q` for json safety.  That could be a nice "make it better than it was" item we could add though.